### PR TITLE
Consistent docs

### PR DESCRIPTION
**Pull request by dhrone**
* Added ``ws0010`` driver to support Winstar WS0010 controller-based devices
* Added ``winstar_weh`` driver to support character versions of devices using WS0010 controllers
* Updated documentation to be consistent with ``luma.lcd`` documentation

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
+are included in :doc:`python-usage`.
 
-Install the latest version of the library directly from PyPI_::
+.. note:: This library has been tested against Python 3.5, 3.6, 3.7 and 3.8.
 
-  $ sudo apt install python-dev python-pip libfreetype6-dev libjpeg-dev build-essential libopenjp2-7 libtiff5
-  $ sudo -H pip install --upgrade luma.oled
+   It was *originally* tested with Raspbian on a rev.2 model B, with a vanilla
+   kernel version 4.1.16+, and has subsequently been tested on Raspberry Pi
+   (Raspbian Jessie, Stretch and Buster) models A, B2, 3B, Zero, Zero W,
```
Comment (thijstriemstra) [inline doc/install.rst:24]: also add pi 4

File: doc/intro.rst [hunk 2]
```diff
@@ -1,11 +1,11 @@
 Introduction
-------------
-Interfacing `OLED matrix displays
+============
+Luma.OLED provides a Python3 interface to `OLED matrix displays
```
Comment (thijstriemstra) [inline doc/intro.rst:3]: lowercase luma.oled

File: doc/intro.rst [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 <https://github.com/rm-hull/luma.oled/wiki/Usage-&-Benchmarking>`_ with the
 SSD1306, SSD1309, SSD1322, SSD1325, SSD1327, SSD1331, SSD1351, SSD1362, SH1106 or
-WS0010 driver in Python 3 using I2C/SPI/Parallel interfaces on the Raspberry Pi
-and other linux-based single-board computers: the library provides a
-Pillow_-compatible drawing canvas, and other functionality to support:
+WS0010 controllers to connect with Raspberry Pi and other linux-based
```
Comment (thijstriemstra) [inline doc/intro.rst:6]: Linux needs capital

File: doc/software.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+-----------------------
+If ``pip`` is unable to automatically install its dependencies you will have to
+add them manually.
+
+``luma.oled`` relies on the following packages.
+
+.. csv-table::
+  :header: "Package", "Description"
+  :widths: 15, 25
+
```
Comment (thijstriemstra) [inline doc/software.rst:23]: see my prev luma.lcd comment about listing the dependencies.

File: doc/software.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    $ sudo apt-get install python3-imaging
+
+  or::
+
+    $ sudo apt-get install python3-pil
+
+Permissions
+-----------
+``Luma.oled`` uses hardware interfaces that require permission to access.  After you
```
Comment (thijstriemstra) [inline doc/software.rst:68]: lowercase

File: setup.py [hunk 6]
```diff
@@ -66,6 +68,7 @@ def find_version(*file_paths):
     namespace_packages=["luma"],
     zip_safe=False,
     install_requires=["luma.core>=1.16.1"],
+
```
Comment (thijstriemstra) [inline setup.py:71]: no need for newline

**Review by thijstriemstra — CHANGES_REQUESTED** [review]

**Review by rm-hull — APPROVED** [review]
👍

**Comment by rm-hull** [comment]
Happy for it to be merged once @thijstriemstra approves

File: setup.py (hunk 6 above)
Comment (thijstriemstra) [reply setup.py:70]: and probably needs to rely on luma.core >= 1.16.2

File: doc/install.rst (hunk 1 above)
Comment (dhrone) [reply doc/install.rst]: Reworded paragraph to make it clearer which hardware and OS versions had been tested.

File: doc/software.rst (hunk 4 above)
Comment (dhrone) [reply doc/software.rst]: I've removed the references and reworded the section both here and in ``luma.lcd``

File: setup.py (hunk 6 above)
Comment (dhrone) [reply setup.py]: Removed newline

File: doc/intro.rst (hunk 2 above)
Comment (dhrone) [reply doc/intro.rst]: Changed as suggested

File: doc/intro.rst (hunk 3 above)
Comment (dhrone) [reply doc/intro.rst]: Changed as suggested

File: doc/software.rst (hunk 5 above)
Comment (dhrone) [reply doc/software.rst]: Changed as suggested

File: doc/software.rst [hunk 7]
```diff
@@ -13,43 +13,19 @@ install them automatically.
...
+add them manually.  To resolve the issues you will need to add the appropriate
+development packages to continue.
 
 If you are using Raspbian Stretch or Raspberry Pi OS (Buster) you should
 be able to use the following commands to add the required packages::
 
 $ sudo apt-get update
 $ sudo apt-get install python3 python3-pip python3-pil libjpeg-dev zlib1g-dev libfreetype6-dev liblcms2-dev libopenjp2-7 libtiff5 -y
-$ sudo -H pip3 install luma.oled
+$ sudo -H pip3 install luma.lcd
```
Comment (thijstriemstra) [inline doc/software.rst:24]: should be oled

File: doc/hardware.rst [hunk 8]
```diff
@@ -159,28 +180,36 @@ OLED Pin   Name   Remarks      RPi Pin  RPi Function
...
+    you can select other pins and pass a ``gpio_DC`` and/or a ``gpio_RST``
     argument specifying the new *BCM* pin numbers in your serial interface create
     call.
 
-  * The use of the terms 4-wire and 3-wire SPI are a bit confusing because, in
+  * The use of the terms 4-wire and 3-wire SPI are a bit confusing because in
     most SPI documentation, the terms are used to describe the regular 4-wire
     configuration of SPI and a 3-wire mode where the input and output lines, MOSI
     and MISO, have been combined into a single line called SISO. However, in the
-    context of these OLED controllers, 4-wire means MOSI + Data/Command and 3-wire
```
Comment (thijstriemstra) [inline doc/hardware.rst:176]: should be OLED

File: doc/software.rst (hunk 7 above)
Comment (dhrone) [reply doc/software.rst:24]: Good catch.

File: doc/hardware.rst (hunk 8 above)
Comment (dhrone) [reply doc/hardware.rst:176]: Changed to OLED

**Review by thijstriemstra — APPROVED** [review]

**Comment by thijstriemstra** [comment]
@dhrone we always squash-merge pull requests, can you also do this next time?

**Comment by dhrone** [comment]
@thijstriemstra Absolutely.  Thanks for the guidance.

**Comment by rm-hull** [comment]
@dhrone happy for you to merge PRs into the main branch, but could you pick the "Squash and merge" option in the future please? 

(It just keeps the main branch clear of tramlines) e.g.:
image

Also, I am about to release 3.6.0 of luma.oled with your changes - is it good to go?
Ditto for latest luma.lcd ?

**Comment by thijstriemstra** [comment]
@rm-hull see https://github.com/rm-hull/luma.oled/pull/298#issuecomment-696084962

**Comment by dhrone** [comment]
I think they are ready to go unless you want to extend the hd44780 documentation.

**Comment by rm-hull** [comment]
apologies both - i did not notice the earlier comment

**Pull request merged**